mods/gen6/random-teams.js

```diff
@@ -61,8 +61,8 @@ class RandomGen6Teams extends RandomTeams {
 			hasAbility[template.abilities['H']] = true;
 		}
 		let availableHP = 0;
-		for (let i = 0, len = movePool.length; i < len; i++) {
-			if (movePool[i].substr(0, 11) === 'hiddenpower') availableHP++;
+		for (const move of movePool) {
+			if (move.substr(0, 11) === 'hiddenpower') availableHP++;
```

Comment:
Also `setMoveid.startsWith('hiddenpower')`, just do it every time this comes up, too.